spokestack/asr/google/speech_recognizer.py

```diff
@@ -0,0 +1,119 @@
+"""
+This module contains the google asr speech recognizer
+"""
+import logging
+from queue import Queue
+from threading import Thread
+from typing import Any
+
+import numpy as np  # type: ignore
+from google.cloud import speech  # type: ignore
+from google.oauth2 import service_account  # type: ignore
+
+from spokestack.context import SpeechContext
+
+
+_LOG = logging.getLogger(__name__)
+
+
+class GoogleSpeechRecognizer:
+    """Transforms speech into text using Google's ASR.
+
+    Args:
+        language (str): language of input audio
+        language (str): language of input audio
+        credentials (optional): google API credentials or path to credentials.
+                                if set to None these will be pulled from the
+                                environment variable: GOOGLE_APPLICATION_CREDENTIALS
+        sample_rate (int): sample rate of the input audio (Hz)
+        **kwargs (optional): additional keyword arguments
+    """
+
+    def __init__(
+        self, language, credentials=None, sample_rate=16000, **kwargs,
+    ) -> None:
+        if credentials:
+            if isinstance(credentials, str):
+                credentials = service_account.Credentials.from_service_account_file(
+                    credentials
+                )
+            else:
```

Comment:
might be good to add an explicit check for `dict` here and raise a valueerror on bad credentials so the user will know what is going on